rclcpp/test/rclcpp/executors/test_executors.cpp

```diff
@@ -221,6 +221,56 @@ TYPED_TEST(TestExecutors, spinWhileAlreadySpinning) {
   executor.remove_node(this->node, true);
 }
 
+// Check if timeout is respected
+TYPED_TEST(TestExecutors, testSpinUntilTimeout) {
+  using ExecutorType = TypeParam;
+  ExecutorType executor;
+  executor.add_node(this->node);
+
+  // block intil timeout is fulfilled.
+  auto start = std::chrono::steady_clock::now();
+  executor.spin_until_timeout(2s);
+  executor.remove_node(this->node, true);
+  // Check if timeout was fulfilled
+  EXPECT_GT(2s, (std::chrono::steady_clock::now() - start));
```

Comment:
how about checking if it returns in specific time window?

e.g) `EXPECT_LT(3s, (std::chrono::steady_clock::now() - start));`